### DISCUSSION TITLE
Same Subnet multible use

### DISCUSSION THREAD

**Discussion by ByteAndBits** [netbox-community/netbox#18005] [Help Wanted!] [answered]
Hi,

we have multible branch offices who use the same subnet.
Can I use same subnets multible times but in different branch offices?

Regards

**Comment by a084ed22** [comment]
You have a few choices to achieve this. You can either disable the uniqueness constraint or use different VRFs to handle different prefixes and IP addresses.

**Comment by candlerb** [comment, marked as answer]
VRFs are what you need here. In Netbox, a VRF is both a layer 3 routing domain and a network namespace (i.e. domain of uniqueness for IP addresses). So even if you're not actually using multiple routing tables within the same network, and hence no actual VRFs, you can create a separate VRF for site A and site B, and put their respective prefixes and addresses within those VRFs, and they will not conflict.

**Reply by PieterL75** [reply]
> a VRF is both a layer 3 routing domain and a network namespace

It would be awesome if this could be split up !

**Comment by jorgenspange** [comment]
@candlerb I am in the process of migrating into netbox now, and my trouble is that under my 10/8 block my subnets are assigned to different vrf's. And it seems that when I assign 10/8 to global and 10.12.13.0/24 to another vrf. Both 10.0.0.0/8 and 10.12.13.14.0/24 is nested at the same level.

Any idea to solving this? I'd like to have everything nested as I the entire range is handled by the same aggregate anyhow.

**Reply by PieterL75** [reply]
It's not possible in netbox..
one vrf is one routing domain.. there is no way to use one prefix in different vrfs.
using the global vrf, is a wierd way of 'fixing' this.. I don't want my private IP space in 'global'. I want to know what prefix can be used in what VRF's

I've tried to get this concept implemented several times, but it never got approved..
There is a need of a higher 'IP Space' that enforces the uniqueness, in stead of relying on the VRF.

I would say, go for it and submit a FR.. I'll upvote for sure !

**Comment by lowinger42** [comment]
On 2025-09-26 16:24, PieterL75 wrote:
> I would say, go for it and submit a FR.. I'll upvote for sure !

I'd also need this feature.

just "borrow" nautobots implementation of namespace

https://docs.nautobot.com/projects/core/en/stable/user-guide/core-data-model/ipam/namespace/


:)


-- 
Regards / Med vänlig Hälsning
Anders Löwinger, Abundo AB, +46 72 2060322
